### PR TITLE
CIF-2129 - remove "id" support for CIF Core Components

### PR DESCRIPTION
**Pull request by mhaack**
## Description

Remove "id" support mainly used for categories for all relevant CIF Core Components. This one is huge, sorry. But it is 98% removing stuff.

Some tests have been reworked a little. "Highlights" with more than just deleted code to review:
* https://github.com/adobe/aem-core-cif-components/pull/588/files#diff-f0a57b2faf7c72e55869260792116b0f26106aa776a76e30c1b993a60e209dd3
* https://github.com/adobe/aem-core-cif-components/pull/588/files#diff-142d9e196ee29be35043977ebed365d585cd263ca8b168e8034ebb7a6f444c3e
* https://github.com/adobe/aem-core-cif-components/pull/588/files#diff-27b27eb193ccc0105be818f374c0f1fc16e923d43c4895e67774629eeebfd26f
* https://github.com/adobe/aem-core-cif-components/pull/588/files#diff-0bcf31af5cbb7c0362413cc00d3c3bbf90d29d6dd1e86b420860a3e04565caab
* https://github.com/adobe/aem-core-cif-components/pull/588/files#diff-1fe2e6005a652368e24f88b713e6dd127521cd3278a2b2329cefb0ac06c0bf45

## Related Issue

CIF-2129

## Motivation and Context

Prepare for CIF 2.0

## How Has This Been Tested?

Adjusted unit tests & ITs. UI tests did not require any adjustment.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java [hunk 1]
```diff
@@ -164,13 +163,16 @@ protected void populate() {
...
-            case UID:
-                return category.getUid().toString();
-            default:
-                return "";
+        if (identifierType.equals(CategoryIdentifierType.UID)) {
+            return category.getUid().toString();
         }
+        return StringUtils.EMPTY;
+
+        // switch (identifierType) {
```
Comment (herzog31) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java:171]: What's with this comment here?

**Review by herzog31 — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java (hunk 1 above)
Comment (mhaack) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java]: Rmoved

**Review by dplaton — APPROVED** [review]

**Comment by mhaack** [comment]
Setting the PR to draft for now as new features have been added to master branch which must be reviewed as well.

**Comment by LSantha** [comment]
I think CommerceContentFragmentImpl was not updated yet.

**Comment by mhaack** [comment]
@LSantha removed `ID` usage from CommerceContentFragmentImpl as well.

**Pull request merged**